docker_test.go

```diff
@@ -509,3 +531,40 @@ func getExpectedFatManifestBuildInfo(t *testing.T) entities.BuildInfo {
 	assert.NoError(t, json.Unmarshal(data, &buildinfo))
 	return buildinfo
 }
+
+func TestNativeDockerPushPull(t *testing.T) {
+	cleanup := initNativeDockerTest(t, corecontainercmds.RtMinVersion, "")
+	defer cleanup()
+	pushBuildNumber := "2"
+	pullBuildNumber := "3"
+	module := "native-docker-module"
+	image, err := inttestutils.BuildTestImage(tests.DockerImageName+":"+pushBuildNumber, "", container.DockerClient)
+	assert.NoError(t, err)
+	// Add docker cli flag '-D' to check we ignore them
+	runNativeDocker(t, "docker", "-D", "push", image, "--build-name="+tests.DockerBuildName, "--build-number="+pushBuildNumber, "--module="+module)
+
+	inttestutils.ValidateGeneratedBuildInfoModule(t, tests.DockerBuildName, pushBuildNumber, "", []string{module}, buildinfo.Docker)
+	runRt(t, "build-publish", tests.DockerBuildName, pushBuildNumber)
+	imagePath := path.Join(*tests.DockerLocalRepo, tests.DockerImageName, pushBuildNumber) + "/"
+	validateContainerBuild(tests.DockerBuildName, pushBuildNumber, imagePath, module, 7, 5, 7, t)
+	inttestutils.DeleteTestImage(t, image, container.DockerClient)
+
+	runNativeDocker(t, "docker", "-D", "pull", image, "--build-name="+tests.DockerBuildName, "--build-number="+pullBuildNumber, "--module="+module)
+	runRt(t, "build-publish", tests.DockerBuildName, pullBuildNumber)
+	imagePath = path.Join(*tests.DockerLocalRepo, tests.DockerImageName, pullBuildNumber) + "/"
+	validateContainerBuild(tests.DockerBuildName, pullBuildNumber, imagePath, module, 0, 7, 0, t)
+	inttestutils.DeleteTestImage(t, image, container.DockerClient)
+
+	inttestutils.ContainerTestCleanup(t, serverDetails, artHttpDetails, tests.DockerImageName, tests.DockerBuildName, *tests.DockerLocalRepo)
```

Comment:
I suggest we put this function as a defer in the beginning of this function. The same goes for the other usage of this function.